native-sql-engine/core/src/main/scala/com/intel/oap/expression/ColumnarTernaryOperator.scala

```diff
@@ -223,6 +223,42 @@ class ColumnarSubstringIndex(strExpr: Expression, delimExpr: Expression,
   }
 }
 
+class ColumnarStringReplace(
+    srcExpr: Expression,
+    searchExpr: Expression,
+    replaceExpr: Expression)
+  extends StringReplace(srcExpr, searchExpr, replaceExpr) with ColumnarExpression {
+
+  buildCheck()
+  def buildCheck(): Unit = {
+    val unsupportedDataType =
+      Seq(srcExpr.dataType, searchExpr.dataType, replaceExpr.dataType)
+        .filterNot(_ == StringType)
+    if (unsupportedDataType.nonEmpty) {
+      throw new UnsupportedOperationException(
+        s"${unsupportedDataType.mkString(",")} is not supported in ColumnarStringReplace.")
+    }
+  }
+
```

Comment:
For expression not supporting whole stage code gen, we need to add the below method. (Some expressions are still lacking this method, but for new expression support, we recommend to follow this convention.)

override def supportColumnarCodegen(args: java.lang.Object): Boolean = {
    false
  }

Reply:
Ah, thanks for your suggestion. I have just checked the code, it is true that replace does not support WSCG. I will add a TODO and implement the corresponding Codegen logic later.